### ISSUE TITLE
 a problem at the beginning of the training

### ISSUE THREAD

**Issue by shawn5549** [MIC-DKFZ/nnUNet#2306]
(nnUnet_2.3.1) D:\nnUNet-master\nnUNet>nnUNetv2_train 14 3d_fullres 0

############################
INFO: You are using the old nnU-Net default plans. We have updated our recommendations. Please consider using those instead! Read more here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/resenc_presets.md
############################

Using device: cuda:0

#######################################################################
Please cite the following paper when using nnU-Net:
Isensee, F., Jaeger, P. F., Kohl, S. A., Petersen, J., & Maier-Hein, K. H. (2021). nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation. Nature methods, 18(2), 203-211.
#######################################################################

2024-06-19 16:08:49.668111: do_dummy_2d_data_aug: False
2024-06-19 16:08:49.672644: Using splits from existing split file: D:\nnUNet-master\nnUNet\nnUNet_preprocessed\Dataset014_faceCT\splits_final.json
2024-06-19 16:08:49.675648: The split file contains 5 splits.
2024-06-19 16:08:49.678646: Desired fold for training: 0
2024-06-19 16:08:49.681648: This split has 4 training and 2 validation cases.
using pin_memory on device 0
using pin_memory on device 0
2024-06-19 16:08:58.863785: Using torch.compile...
C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\optim\lr_scheduler.py:28: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn("The verbose parameter is deprecated. Please use get_last_lr() "

This is the configuration used by this training:
Configuration name: 3d_fullres
 {'data_identifier': 'nnUNetPlans_3d_fullres', 'preprocessor_name': 'DefaultPreprocessor', 'batch_size': 2, 'patch_size': [128, 128, 112], 'median_image_size_in_voxels': [768.0, 768.0, 576.0], 'spacing': [1.0, 1.0, 1.0], 'normalization_schemes': ['CTNormalization'], 'use_mask_for_norm': [False], 'resampling_fn_data': 'resample_data_or_seg_to_shape', 'resampling_fn_seg': 'resample_data_or_seg_to_shape', 'resampling_fn_data_kwargs': {'is_seg': False, 'order': 3, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_seg_kwargs': {'is_seg': True, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_probabilities': 'resample_data_or_seg_to_shape', 'resampling_fn_probabilities_kwargs': {'is_seg': False, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'architecture': {'network_class_name': 'dynamic_network_architectures.architectures.unet.PlainConvUNet', 'arch_kwargs': {'n_stages': 6, 'features_per_stage': [32, 64, 128, 256, 320, 320], 'conv_op': 'torch.nn.modules.conv.Conv3d', 'kernel_sizes': [[3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3]], 'strides': [[1, 1, 1], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 1]], 'n_conv_per_stage': [2, 2, 2, 2, 2, 2], 'n_conv_per_stage_decoder': [2, 2, 2, 2, 2], 'conv_bias': True, 'norm_op': 'torch.nn.modules.instancenorm.InstanceNorm3d', 'norm_op_kwargs': {'eps': 1e-05, 'affine': True}, 'dropout_op': None, 'dropout_op_kwargs': None, 'nonlin': 'torch.nn.LeakyReLU', 'nonlin_kwargs': {'inplace': True}, 'deep_supervision': True}, '_kw_requires_import': ['conv_op', 'norm_op', 'dropout_op', 'nonlin']}, 'batch_dice': True}

These are the global plan.json settings:
 {'dataset_name': 'Dataset014_faceCT', 'plans_name': 'nnUNetPlans', 'original_median_spacing_after_transp': [1.0, 1.0, 1.0], 'original_median_shape_after_transp': [768, 768, 576], 'image_reader_writer': 'SimpleITKIO', 'transpose_forward': [0, 1, 2], 'transpose_backward': [0, 1, 2], 'experiment_planner_used': 'ExperimentPlanner', 'label_manager': 'LabelManager', 'foreground_intensity_properties_per_channel': {'0': {'max': 1836.0, 'mean': 55.032779693603516, 'median': -52.0, 'min': -1000.0, 'percentile_00_5': -1000.0, 'percentile_99_5': 1560.0, 'std': 561.9766845703125}}}

2024-06-19 16:08:59.048910: unpacking dataset...
2024-06-19 16:08:59.290647: unpacking done...
2024-06-19 16:08:59.295252: Unable to plot network architecture: nnUNet_compile is enabled!
2024-06-19 16:08:59.311333:
2024-06-19 16:08:59.314892: Epoch 0
2024-06-19 16:08:59.318892: Current learning rate: 0.01
Traceback (most recent call last):
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\Scripts\nnUNetv2_train.exe\__main__.py", line 7, in <module>
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\nnunetv2\run\run_training.py", line 274, in run_training_entry
    run_training(args.dataset_name_or_id, args.configuration, args.fold, args.tr, args.p, args.pretrained_weights,
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\nnunetv2\run\run_training.py", line 210, in run_training
    nnunet_trainer.run_training()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\nnunetv2\training\nnUNetTrainer\nnUNetTrainer.py", line 1295, in run_training
    train_outputs.append(self.train_step(next(self.dataloader_train)))
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\nnunetv2\training\nnUNetTrainer\nnUNetTrainer.py", line 922, in train_step
    output = self.network(data)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\eval_frame.py", line 451, in _fn
    return fn(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 921, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state, skip=1)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 786, in _convert_frame
    result = inner_convert(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 400, in _convert_frame_assert
    return _compile(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 676, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 535, in compile_inner
    out_code = transform_code_object(code, transform)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\bytecode_transformation.py", line 1036, in transform_code_object
    transformations(instructions, code_options)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 165, in _fn
    return fn(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\convert_frame.py", line 500, in transform
    tracer.run()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\symbolic_convert.py", line 2149, in run
    super().run()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\symbolic_convert.py", line 810, in run
    and self.step()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\symbolic_convert.py", line 773, in step
    getattr(self, inst.opname)(inst)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\symbolic_convert.py", line 2268, in RETURN_VALUE
    self.output.compile_subgraph(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\output_graph.py", line 1001, in compile_subgraph
    self.compile_and_call_fx_graph(tx, pass2.graph_output_vars(), root)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\output_graph.py", line 1178, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\output_graph.py", line 1251, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\output_graph.py", line 1232, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\repro\after_dynamo.py", line 117, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\__init__.py", line 1731, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\compile_fx.py", line 1330, in compile_fx
    return aot_autograd(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\backends\common.py", line 58, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_functorch\aot_autograd.py", line 903, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_functorch\aot_autograd.py", line 628, in create_aot_dispatcher_function
    compiled_fn = compiler_fn(flat_fn, fake_flat_args, aot_config, fw_metadata=fw_metadata)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_functorch\_aot_autograd\runtime_wrappers.py", line 443, in aot_wrapper_dedupe
    return compiler_fn(flat_fn, leaf_flat_args, aot_config, fw_metadata=fw_metadata)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_functorch\_aot_autograd\runtime_wrappers.py", line 648, in aot_wrapper_synthetic_base
    return compiler_fn(flat_fn, flat_args, aot_config, fw_metadata=fw_metadata)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_functorch\_aot_autograd\jit_compile_runtime_wrappers.py", line 352, in aot_dispatch_autograd
    compiled_fw_func = aot_config.fw_compiler(fw_module, adjusted_flat_args)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\compile_fx.py", line 1257, in fw_compiler_base
    return inner_compile(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\repro\after_aot.py", line 83, in debug_wrapper
    inner_compiled_fn = compiler_fn(gm, example_inputs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\debug.py", line 304, in inner
    return fn(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\compile_fx.py", line 438, in compile_fx_inner
    compiled_graph = fx_codegen_and_compile(
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\compile_fx.py", line 714, in fx_codegen_and_compile
    compiled_fn = graph.compile_to_fn()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\graph.py", line 1307, in compile_to_fn
    return self.compile_to_module().call
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\graph.py", line 1250, in compile_to_module
    self.codegen_with_cpp_wrapper() if self.cpp_wrapper else self.codegen()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\graph.py", line 1205, in codegen
    self.scheduler = Scheduler(self.buffers)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_dynamo\utils.py", line 262, in time_wrapper
    r = func(*args, **kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 1267, in __init__
    self.nodes = [self.create_scheduler_node(n) for n in nodes]
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 1267, in <listcomp>
    self.nodes = [self.create_scheduler_node(n) for n in nodes]
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 1358, in create_scheduler_node
    return SchedulerNode(self, node)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 687, in __init__
    self._compute_attrs()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 698, in _compute_attrs
    group_fn = self.scheduler.get_backend(self.node.get_device()).group_fn
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 2276, in get_backend
    self.backends[device] = self.create_backend(device)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\torch\_inductor\scheduler.py", line 2268, in create_backend
    raise RuntimeError(
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
RuntimeError: Cannot find a working triton installation. More information on installing Triton can be found at https://github.com/openai/triton

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information

You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True

Exception in thread Thread-1 (results_loop):
Traceback (most recent call last):
Exception in thread Thread-2 (results_loop):
Traceback (most recent call last):
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\threading.py", line 1016, in _bootstrap_inner
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\threading.py", line 953, in run
    self.run()
    self._target(*self._args, **self._kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 125, in results_loop
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\threading.py", line 953, in run
    raise e
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 103, in results_loop
    self._target(*self._args, **self._kwargs)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 125, in results_loop
    raise RuntimeError("One or more background workers are no longer alive. Exiting. Please check the "
    raise e
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 103, in results_loop
RuntimeError: One or more background workers are no longer alive. Exiting. Please check the print statements above for the actual error message
    raise RuntimeError("One or more background workers are no longer alive. Exiting. Please check the "
RuntimeError: One or more background workers are no longer alive. Exiting. Please check the print statements above for the actual error message
Exception ignored in: <function NonDetMultiThreadedAugmenter.__del__ at 0x000001D393FA3520>
Traceback (most recent call last):
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 253, in __del__
    self._finish()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 238, in _finish
    self.abort_event.set()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 338, in set
    self._cond.notify_all()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 297, in notify_all
    self.notify(n=sys.maxsize)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 272, in notify
    assert not self._wait_semaphore.acquire(
OSError: [WinError 6] 句柄无效。
Exception ignored in: <function NonDetMultiThreadedAugmenter.__del__ at 0x000001D393FA3520>
Traceback (most recent call last):
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 253, in __del__
    self._finish()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\site-packages\batchgenerators\dataloading\nondet_multi_threaded_augmenter.py", line 238, in _finish
    self.abort_event.set()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 338, in set
    self._cond.notify_all()
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 297, in notify_all
    self.notify(n=sys.maxsize)
  File "C:\Users\user\.conda\envs\nnUnet_2.3.1\lib\multiprocessing\synchronize.py", line 272, in notify
    assert not self._wait_semaphore.acquire(
OSError: [WinError 6] 句柄无效。

**Comment by shawn5549** [comment]
(nnUnet_2.3.1) D:\nnUNet-master\nnUNet>pip list
Package                       Version
----------------------------- ------------
acvl_utils                    0.2
batchgenerators               0.25
certifi                       2024.6.2
charset-normalizer            3.3.2
colorama                      0.4.6
connected-components-3d       3.16.1
contourpy                     1.2.1
cycler                        0.12.1
dicom2nifti                   2.4.11
dynamic_network_architectures 0.3.1
filelock                      3.13.1
fonttools                     4.53.0
fsspec                        2024.2.0
future                        1.0.0
gmpy2                         2.1.2
graphviz                      0.20.3
idna                          3.7
imagecodecs                   2024.6.1
imageio                       2.34.1
intel-openmp                  2021.4.0
Jinja2                        3.1.3
joblib                        1.4.2
kiwisolver                    1.4.5
labelImg                      1.8.6
lazy_loader                   0.4
linecache2                    1.0.0
MarkupSafe                    2.1.3
matplotlib                    3.9.0
mkl                           2021.4.0
mkl-fft                       1.3.1
mkl-random                    1.2.2
mkl-service                   2.4.0
mpmath                        1.3.0
networkx                      3.2.1
nibabel                       5.2.1
nnunetv2                      2.4.2
numpy                         1.24.3
packaging                     24.1
pandas                        2.2.2
pillow                        10.3.0
pip                           24.0
pydicom                       2.4.4
pyparsing                     3.1.2
PyQt5                         5.15.9
pyqt5-plugins                 5.15.9.2.3
PyQt5-Qt5                     5.15.2
pyqt5-tools                   5.15.9.3.3
python-dateutil               2.9.0.post0
python-dotenv                 1.0.0
python-gdcm                   3.0.24.1
pytz                          2024.1
PyYAML                        6.0.1
qt5-applications              5.15.2.2.3
qt5-tools                     5.15.2.1.3
requests                      2.32.3
scikit-image                  0.23.2
scikit-learn                  1.5.0
scipy                         1.13.1
seaborn                       0.13.2
setuptools                    69.5.1
SimpleITK                     2.3.1
six                           1.16.0
sympy                         1.12
tbb                           2021.11.0
threadpoolctl                 3.5.0
tifffile                      2024.5.22
torch                         2.3.1+cu118
torchaudio                    2.3.1+cu118
torchvision                   0.18.1+cu118
tqdm                          4.66.4
traceback2                    1.4.0
typing_extensions             4.11.0
tzdata                        2024.1
unittest2                     1.1.0
urllib3                       2.2.2
wheel                         0.43.0
yacs                          0.1.8

**Comment by shawn5549** [comment]
I tried using a small public data set (26MB), 2d, and got the same error

**Comment by shawn5549** [comment]
句柄无效  is  ‘’The handle is invalid ‘’

**Comment by JackRio** [comment]
I had something similar. This solution worked for me.

https://github.com/pytorch/pytorch/issues/119054

TLDR:
```
pip install nvidia-cuda-nvcc-cu11
export TRITON_PTXAS_PATH=<path-to-python-version>/dist-packages/nvidia/cuda_nvcc/bin/ptxas
Or add the above path to the bashrc (Linux)
```

**Comment by shawn5549** [comment]
thank you , I tred,but it doesn't work for me

**Comment by chris-rapson-formus** [comment]
I'm seeing the same error, and @JackRio's suggestion also didn't help me.

When I changed my base docker image to use a newer CUDA version (`pytorch/pytorch:2.3.1-cuda12.1-cudnn8-runtime`), this error is replaced by another one:

```
RuntimeError: The NVIDIA driver on your system is too old (found version 11040). Please update your GPU driver by downloading"
```

Both errors are hard to debug, because they're not reproducible on my local system. They only come up when I try to run training in the cloud.

**Comment by GregorKoehler** [comment]
Hi @shawn5549 and others,

a few things to note here:

* Please don't just post error messages as the github issue. The issues aren't meant to offload debugging work to others. Please provide some more context in the future.
* The error message points to a missing Triton installation: `RuntimeError: Cannot find a working triton installation. More information on installing Triton can be found at https://github.com/openai/triton` - Have you tried pip installing triton?
* As the error in general seems to be connected to torch compile. If you are okay with not using torch compile in nnUNet, you can try setting the environment variable `nnUNet_compile` to `nnUNet_compile=False` when running the training.

**Comment by chris-rapson-formus** [comment]
Thanks for the input @GregorKoehler. `nnUNet_compile=False` was what I ended up going with. I think Fabian suggested that on another issue that was related to `torch._dynamo`. I miss out on some of the performance improvements, but it does run.

My docker image does have triton installed, because it calls the `setup.py` script in the nnunet repo. For some reason, it just can't find triton when it needs it:
```
# pip freeze | grep -i triton
triton==2.3.1
```

**Comment by GregorKoehler** [comment]
Hi @chris-rapson-formus,

thanks for the update! Have you already tried the suggested

> Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information

as proposed in the error log?

**Comment by GregorKoehler** [comment]
Hi @chris-rapson-formus,

thanks for the update, with triton being quite new, this might indeed be the case. It might be worth it to update nnUNet to the current master regularly.

**Comment by GregorKoehler** [comment]
You might also want to check out this potentially related issue: https://github.com/MIC-DKFZ/nnUNet/issues/2330 . Apparently, triton does not ship with the windows installation of PyTorch. Also, can you confirm that your hardware is compatible with triton? (see https://github.com/triton-lang/triton?tab=readme-ov-file#compatibility)

**Comment by chris-rapson-formus** [comment]
I was using an A100 in the google cloud when I saw this error. I couldn't reproduce it when running on an on-prem 1070. My container was built on a linux base image. I don't think it's hardware or OS in my case.

There are so many other components though. CUDA, torch, numpy, python and even nnunet can all have different versions. I'm only slightly surprised that triton isn't compatible with all of the various versions of each of those packages.

**Comment by GregorKoehler** [comment]
Thanks for the info! As going with `nnUNet_compile=False` resolved your issue and the issue OP isn't responding since a while, I'll close this issue for now.
